packages/react-vapor/src/components/table-hoc/utils/TableHooks.tsx

```diff
@@ -0,0 +1,20 @@
+import * as React from 'react';
+
+export const useCustomLayoutEffect = (isLoading: boolean) => {
+    const tableHeaderRef = React.useRef<HTMLTableHeaderCellElement>();
+    const [columnWidth, setColumnWidth] = React.useState<number>();
+
+    React.useLayoutEffect(() => {
+        if (tableHeaderRef.current && !isLoading) {
+            setColumnWidth(tableHeaderRef.current.offsetWidth);
+        }
+    }, [isLoading]);
+
+    const style = {
+        width: `${columnWidth}px`,
+        minWidth: `${columnWidth}px`,
+        maxWidth: `${columnWidth}px`,
+    };
```

Comment:
What do you think about only returning the style during the loading, an empty object otherwise?